src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -0,0 +1,159 @@
+    loginList: PropTypes.shape({
+        /** The partner creating the account. It depends on the source: website, mobile, integrations, ... */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** The date when the login was validated, used to show the brickroad status */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+
+        /** Field-specific pending states for offline UI status */
+        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    betas: [],
+    loginList: {},
+};
+
+class NewContactMethodPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            login: '',
+            password: '',
+        };
+        this.onLoginChange = this.onLoginChange.bind(this);
+        this.validateForm = this.validateForm.bind(this);
+        this.submitForm = this.submitForm.bind(this);
+    }
+
+    onLoginChange(login) {
+        this.setState({login});
+    }
+
+    /**
+    * Determine whether the form is valid
+    *
+    * @returns {Boolean}
+    */
+    validateForm() {
+        const login = this.state.login.trim();
+        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login);
+
+        return (Permissions.canUsePasswordlessLogins(this.props.betas) || this.state.password)
+            && (Str.isValidEmail(login) || Str.isValidPhone(phoneLogin));
+    }
+
+    submitForm() {
+        // If this login already exists, just go back.
+        if (lodashGet(this.props.loginList, this.state.login)) {
+            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
+            return;
+        }
+        User.addNewContactMethodAndNavigate(this.state.login, this.state.password);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper
+                onTransitionEnd={() => {
+                    if (!this.loginInputRef) {
+                        return;
+                    }
+                    this.loginInputRef.focus();
+                }}
+            >
+                <HeaderWithCloseButton
+                    title={this.props.translate('contacts.newContactMethod')}
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS)}
+                    onCloseButtonPress={() => Navigation.dismissModal(true)}
+                />
+                <ScrollView>
+                    <Text style={[styles.ph5, styles.mb5]}>
+                        {this.props.translate('common.pleaseEnterEmailOrPhoneNumber')}
+                    </Text>
+                    <View style={[styles.ph5, styles.mb6]}>
+                        <TextInput
```

Comment:
👋 This has caused a small regression in https://github.com/Expensify/App/issues/18003
Since this is an email/phone TextInput, it should've had the `email` keyboard type